Controller/ProcessController.php

```diff
@@ -392,6 +393,12 @@ public function complete(Request $request, Process $process, ServiceRepository $
             $process->setCompletedAt($completedAt);
             $process->setLastCompletedAt($completedAt);
 
+            // If it's the first time the process is completed,
+            // we set the originally completed date.
+            if (null === $process->getLastReopened()) {
+                $process->setOriginallyCompletedAt($completedAt);
+            }
+
```

Comment:
I'm not sure if this is semantically the same, but this might be easier to understand if this block if moved above

```
$process->setCompletedAt($completedAt);
$process->setLastCompletedAt($completedAt);
```

and rewritten to

```
if (null === $process->getCompletedAt()) {
    $process->setOriginallyCompletedAt($completedAt);
}
```

It may even make sense to move it into `Process:setCompletedAt`:

```
function setCompletedAt($completedAt) {
    if (null === $this->completedAt) {
        $this->originallyCompletedAt = $completedAt;
    }
    $this->completedAt = $completedAt;
}
```

Reply:
CompletedAt will be null if the process have been resumed, and we're only interested in setting the originallyCompletedAt property if the process haven't been resumed before.